docs/sources/reference/config-blocks/import.file.md

```diff
@@ -56,11 +74,71 @@ declare "add" {
 
 {{< /collapse >}}
 
-{{< collapse title="importer.alloy" >}}
+
+This example imports a module from a file inside of a module that is imported via [import.git][]:
+
+{{< collapse title="main.alloy" >}}
+
+```alloy
+import.git "math" {
+  repository = "https://github.com/wildum/module.git"
+  path       = "relative_math.alloy"
+}
+
+math.add "default" {
+  a = 15
+  b = 45
+}
+```
+
+{{< /collapse >}}
+
+{{< collapse title="relative_math.alloy" >}}
+
+```alloy
+import.file "lib" {
+  filename = file.path_join(module_path, "lib.alloy")
+}
```

Comment:
Instead of the keyword, could we introduce a rule that imports from inside a module will be relative to that module's location?

In this example if you have `filename = "./lib.alloy"` it would automatically be relative to the checkout of the git repo. Users can still use absolute paths if they want a different behaviour.

Reply:
this is proposal 4 in the design doc: https://docs.google.com/document/d/16vRoiIxQVVlqqEGngm27Kcun6hF_BTpY2tPOSt05JWU/edit#heading=h.qc2hfy5h8d7z
There are a few cons with this approach and consensus was reached with another approach but nothing is set in stone. If you think that the proposal should be re-evaluated we can discuss it again

Reply:
Yeah, Proposal 4 is a design that I would personally prefer, but it's true that it could break some users. So it'd be something to do before 1.0, but now we have to maintain the existing behaviour. 

We could add a ticket for 2.0 to reconsider this.

I agree that the option you're implementing here is the next best thing.

Reply:
opened an issue to reconsider our decision with a 2.0 label: https://github.com/grafana/alloy/issues/1752